src/rgw/rgw_op.cc

```diff
@@ -4821,7 +4824,8 @@ void RGWDeleteObj::execute(optional_yield y)
         }
       }
     } else {
-      attrs = s->object->get_attrs();
+      attrs = std::move(astate->attrset);
```

Comment:
i don't think we should mutate this `RGWObjState` that's cached in `RGWObjectCtx` here; we'll read it again later in the op

Reply:
ok. but to avoid copying, can i cache a pointer to the attribute set?

Reply:
as long as you don't access that pointer after a potential invalidate.. `del_op->delete_obj(this, y);` could invalidate, and maybe `s->object->swift_versioning_restore(s->obj_ctx, ver_restored, this);` as well